### PR TITLE
allowing custom queries to return more than 100 timeseries

### LINKED ISSUE #48: Allow custom queries to produce more than 100 timeseries
Currently, we get an error saying that the limit for items is 100, which is not very useful to a user as they have little control over this. 
We should be able to create additional `/dataquery` calls with a max of 100 each

### PR DESCRIPTION
**Pull request by hulien22**
There is currently a limit for `/dataquery` to only contain 100 items per request, but this should not limit a user
Now checking if a request will contain more than 100 items and if it does, split it up into more requests.
Fixes #48

### CODE REVIEW

**Comment by martincognite** [comment]
Haven't reviewed yet, but I've heard that 10 is the optimal number of time series to ask for in dataquery. If we're shipping requests in parallell anyway I mean.

**Comment by cognite-cicd** [comment]
[pr-bot]
Run this build with `docker run --rm -it -p 3000:3000 cognite/grafana-cdp-dev:pr-49`

File: src/datasource.ts [hunk 1]
```diff
@@ -168,61 +168,70 @@ export default class CogniteDatasource {
...
-        const idRegex = /\[.*?\]/g; // look for [something]
-        for (const q of queryList) {
-          const matches = q.function.match(idRegex);
-          if (!matches) break;
-          const idsObj = {};
-          for (const match of matches) {
-            idsObj[match.substr(1, match.length - 2)] = true;
+
+      // /dataquery is limited to 100 items, so we need to add new calls if we go over 100 items
+      while (queryList.length > 0) {
```
Comment (martincognite) [inline src/datasource.ts]: lodash has a chunk function, which will split the querylist into objects of `n`. Might be easier to debug than to constantly mutate the running object.

File: src/datasource.ts [hunk 2]
```diff
@@ -168,61 +168,70 @@ export default class CogniteDatasource {
...
+          if (queryReq.aggregates) {
+            target.error =
+              '[ERROR] To use aggregations with functions, use [ID,aggregation,granularity] or [ID,aggregation]';
+            break;
+          }
+          let ids = 0;
+          const idRegex = /\[.*?\]/g; // look for [something]
+          for (const q of slicedQl) {
+            const matches = q.function.match(idRegex);
+            if (!matches) break;
```
Comment (martincognite) [inline src/datasource.ts]: Do you intentionally break here? Or do you want to `continue` to the next `q`.

File: src/datasource.ts [hunk 3]
```diff
@@ -168,61 +168,70 @@ export default class CogniteDatasource {
...
-          ids += Object.keys(idsObj).length;
         }
-        if (ids === 0) ids = 1; // will fail anyways, just show the api error message
+        if (target.assetQuery && target.assetQuery.func && target.tab === Tab.Custom) {
+          if (queryReq.aggregates) {
+            target.error =
+              '[ERROR] To use aggregations with functions, use [ID,aggregation,granularity] or [ID,aggregation]';
+            break;
+          }
+          let ids = 0;
```
Comment (martincognite) [inline src/datasource.ts]: Would be nice to know that this isn't actually `ids`, it's a `idsCount` or similarly. Becomes extra tricky further down, with `/ slicedQl.length` and `/ ids` (and not `/ ids.length`)

File: src/datasource.ts (hunk 2 above)
Comment (hulien22) [reply src/datasource.ts]: Each of these queries all belong to the same query so they would all have the same function. So if one of them doesn't have an id in its function, then none of them do

File: src/datasource.ts (hunk 3 above)
Comment (hulien22) [reply src/datasource.ts]: makes sense, will change

File: src/datasource.ts (hunk 1 above)
Comment (hulien22) [reply src/datasource.ts]: yeah that does sound simpler, I'll fix that

**Comment by cognite-cicd** [comment]
[pr-bot]
Run this build with `docker run --rm -it -p 3000:3000 cognite/grafana-cdp-dev:pr-49`

**Comment by hulien22** [comment]
I tried splitting into chunks of 10, 50 and 100 - still seems like 100 had the best performance (probably because we are being limited by the number of calls we can send in parallel?) 
Something to note is that since limits affect the entire request, having smaller chunk sizes could result in more datapoints for some requests.

**Comment by cognite-cicd** [comment]
[pr-bot]
Run this build with `docker run --rm -it -p 3000:3000 cognite/grafana-cdp-dev:pr-49`

**Pull request merged**